CHANGELOG.md

```diff
@@ -45,6 +45,7 @@
 * [#3673](https://github.com/bbatsov/rubocop/issues/3673): Fix regression on `Style/RedundantSelf` when assigning to same local variable. ([@bankair][])
 * [#4047](https://github.com/bbatsov/rubocop/issues/4047): Allow `find_zone` and `find_zone!` methods in `Rails/TimeZone`. ([@attilahorvath][])
 * [#3457](https://github.com/bbatsov/rubocop/issues/3457): Clear a warning and prevent new warnings. ([@mikegee][])
+* [#4066](https://github.com/bbatsov/rubocop/issues/4066): Register an offense in `Lint/ShadowedException` when an exception is shadowed and there is an implicit begin. ([@rrosenblum][])
```

Comment:
This commit is doing 2 things. The refactor was nice so that we have a common way of checking for this. The need for the refactor came about because a bug was identified in `ShadowedException`. From a user's point of view, this fixes a bug. I will add the issue number to the commit message for tracking as well.